File: MAP/Lab3/src/Model/IfStmt.java
package Model;

import Model.Expression.Expression;
import Model.Statement.*;
import Exception.DivByZeroException;
import Exception.EvaluationException;
import Exception.InexistentSymbolException;


public class IfStmt implements Statement {

    private Expression exp;
    private Statement thenS;
    private Statement elseS;

    public IfStmt(Expression _exp, Statement _thenS, Statement _elseS) {
        exp = _exp;
        thenS = _thenS;
        elseS = _elseS;
    }

    @Override
    public PrgState execute(PrgState state) {
        IDictionary<String, Integer> dict = state.getSymbolT();
        IExecStack<Statement> stack = state.getExecStack();
        try
        {
            int val = exp.Eval(dict);

            if (val != 0)
                stack.push(thenS);
            else
                stack.push(elseS);

            return state;
        }
        catch( EvaluationException | DivByZeroException | InexistentSymbolException e)
        {
            System.out.println(e.getMessage());
            return state;
        }
    }

    @Override
    public String toString() {
        return " IF(" + exp + ")THEN(" + thenS + ")ELSE(" + elseS + ") ";
    }
}

File: MAP/LAB7-MAP/LAB7-MAP/Statements/IfStmt.cs
using MAP_Lab7.Statements;
using MAP_Lab7.Expressions;
using MAP_Lab7.Exceptions;
using System;

namespace MAP_Lab7.Model
{
    public class IfStmt : Statement
    {
        private Statement then;
        private Statement els;
        private IExpression expr;

        public IfStmt(IExpression expre, Statement t, Statement e)
        {
            then = t;
            els = e;
            expr = expre;
        }

        public PrgState Execute(PrgState state)
        {
            try
            {
                if (expr.Eval(state.Dict) != 0)
                    state.ExeStack.PushS(then);
                else
                    state.ExeStack.PushS(els);

                return state;
            }
            catch (DivByZeroException e)
            {
                Console.WriteLine(e.Message);
            }
            catch (EvaluationException e2)
            {
                Console.WriteLine(e2.Message);
            }
            catch (InexistentSymbolException e3)
            {
                Console.WriteLine(e3.Message);
            }
            return state;
        }

        public override string ToString()
        {
            return "IF (" + expr + ") THEN (" + then + ") ELSE (" + els + ")";
        }
    }
}
